pretty_midi/pretty_midi.py

```diff
@@ -186,8 +186,8 @@ def _load_metadata(self, midi_data):
         # signature changes, and lyrics
         self.key_signature_changes = []
         self.time_signature_changes = []
-        self.lyrics = []
-        self.text_events = []
+        self.lyrics = {}
```

Comment:
We don't want to keep parallel lists of lyrics and tests events in both the PrettyMIDI objects and the Instrument objects it contains. In other words, the PrettyMIDI object should not have a lyrics or text_events attribute. These should only be populated when loading from the MIDI file, and then they should be passed on to the Instrument, and the PrettyMIDI object should forget about them. In any PrettyMIDI class method that accesses lyrics or text events, the lyrics and text events of the constitutuent Instruments should be grabbed instead. The logic for populating the lyrics should then go in the __load_instruments method.